tests/DCPS/XTypes/XTypes.h

```diff
@@ -66,5 +69,37 @@ bool check_inconsistent_topic_status(Topic_var topic)
   return true;
 }
 
+void append(DDS::PropertySeq& props, const char* name, const char* value)
+{
+  const DDS::Property_t prop = { name, value, false /*propagate*/ };
+  const unsigned int len = props.length();
+  props.length(len + 1);
+  props[len] = prop;
+}
+
+void create_participant(const DomainParticipantFactory_var& dpf, DomainParticipant_var& dp) {
+  DDS::DomainParticipantQos part_qos;
+
+  if (security_enabled) {
+#if defined(OPENDDS_SECURITY)
+    dpf->get_default_participant_qos(part_qos);
+
+    if (TheServiceParticipant->get_security()) {
+      using namespace DDS::Security::Properties;
+      DDS::PropertySeq& props = part_qos.property.value;
 
+      append(props, AuthIdentityCA, "file:../../security/certs/identity/identity_ca_cert.pem");
+      append(props, AuthIdentityCertificate, "file:../../security/certs/identity/test_participant_02_cert.pem");
+      append(props, AuthPrivateKey, "file:../../security/certs/identity/test_participant_02_private_key.pem");
+      append(props, AccessPermissionsCA, "file:../../security/certs/permissions/permissions_ca_cert.pem");
+      append(props, AccessGovernance, "file:../../security/attributes/governance/governance_AU_UA_ND_NL_NR_signed.p7s");
+      append(props, AccessPermissions, "file:../../security/attributes/permissions/permissions_test_participant_02_allowall_signed.p7s");
+    }
+
+    dp = dpf->create_participant(0, part_qos, 0, DEFAULT_STATUS_MASK);
+#endif
+  } else {
+    dp = dpf->create_participant(0, PARTICIPANT_QOS_DEFAULT, 0, DEFAULT_STATUS_MASK);
+  }
```

Comment:
```suggestion
  }
#endif
  dp = dpf->create_participant(0, part_qos, 0, DEFAULT_STATUS_MASK);
```

Reply:
My intention was to leave dp at 0, if security_enabled is true (should rather call it security_requested), but `OPENDDS_SECURITY` is not defined, so that the caller reports an error and exits.

Reply:
On the other hand that raises a question: how should the test / test script behave on the systems built without security?...

Reply:
I don't know why you'd need to do that. Security should be controlled separately through the `TheServiceParticipant` setting like I mention before. That's not what I really care about though. What I'm trying to get changed though is that you have two `create_participant` statements when you only need one.

Reply:
Fred, I think I have understood your intention and have corrected it in the latest version. There is only one `create_participant` statement in it. Can you please re-check?

Reply:
Yes, I missed that.